### ISSUE TITLE
./install.sh does not work

### ISSUE THREAD

**Issue by mrsoliman** [Blaok/tapa#2]
Using `./install.sh` does not work as the binaries are not longer available on https://about.blaok.me/tapa.

**Comment by Blaok** [comment]
Hi @mrsoliman, could you post the error messages that you see? [CI](https://github.com/Blaok/tapa/actions/workflows/install.yml) suggests the script is working fine.

The binary packages are not directly visible from https://about.blaok.me/tapa as they are organized as a Debian package repository. If you would like to download and install the packages directly, you can take a look at the [gh-pages branch](https://github.com/Blaok/tapa/tree/gh-pages/pool/main).

**Comment by mrsoliman** [comment]
@Blaok Thanks for the explanation.
I am getting this error on Ubuntu 16.04.7 LTS:
`E: The repository 'https://about.blaok.me/fpga-runtime xenial Release' does not have a Release file.`

**Comment by Blaok** [comment]
@mrsoliman CI suggests that the `install.sh` script works fine on a fresh Ubuntu 16.04 container. Can you post the complete output of `sudo apt update` and the content of `/etc/apt/sources.list.d/frt.list`?

**Comment by mrsoliman** [comment]
@Blaok this is from the output of `sudo apt update`:
```
...
Ign:4 https://about.blaok.me/fpga-runtime xenial InRelease                                                                                                                             
...
Ign:7 https://about.blaok.me/tapa xenial InRelease                                                                                                                                     
...
Err:11 https://about.blaok.me/fpga-runtime xenial Release                                                                                                                              
  The HTTP server sent an invalid reply header
...
Err:13 https://about.blaok.me/tapa xenial Release                                                                                                                                 
  The HTTP server sent an invalid reply header
...
Reading package lists... Done                      
W: Target Packages (Packages) is configured multiple times in /etc/apt/sources.list.d/sbt.list:1 and /etc/apt/sources.list.d/sbt.list:2
W: Target Translations (en) is configured multiple times in /etc/apt/sources.list.d/sbt.list:1 and /etc/apt/sources.list.d/sbt.list:2
E: The repository 'https://about.blaok.me/fpga-runtime xenial Release' does not have a Release file.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.
N: See apt-secure(8) manpage for repository creation and user configuration details.
E: The repository 'https://about.blaok.me/tapa xenial Release' does not have a Release file.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.
N: See apt-secure(8) manpage for repository creation and user configuration details.
W: Target Packages (Packages) is configured multiple times in /etc/apt/sources.list.d/sbt.list:1 and /etc/apt/sources.list.d/sbt.list:2
W: Target Translations (en) is configured multiple times in /etc/apt/sources.list.d/sbt.list:1 and /etc/apt/sources.list.d/sbt.list:2
```
And this is the content of `/etc/apt/sources.list.d/frt.list`:
```
deb [arch=amd64] https://about.blaok.me/fpga-runtime xenial main
```

**Comment by Blaok** [comment]
>     The HTTP server sent an invalid reply header

@mrsoliman This should be the cause of problem. Can you check if you can get proper output directly, e.g., running `curl https://about.blaok.me/fpga-runtime/dists/xenial/InRelease` ? Are you connecting through a proxy? [This page](https://forum.hestiacp.com/t/debian-buster-new-install-error/1229/3) suggests that you may need to `sudo apt install apt-transport-https`. If that did not work, [this page](https://support.google.com/chromebook/thread/90817164?hl=en) suggests that a mixture of repos from different OS versions may cause this problem, so you may want to check the output of `cat /etc/apt/sources.list /etc/apt/sources.list.d/*`.

**Comment by Blaok** [comment]
@aanair3 Thanks for your feedback. Could you provide more information about your environment? I would like to reproduce and fix the problem.

**Comment by Blaok** [comment]
It seems that this problem happens when the installation script is interrupted after TAPA's repo is added but before TAPA's key is added. The package management system can be stuck in an inconsistent state. This is now fixed by 197a6718cc60565abf2f9996e8152ba92cf6c01f, which will not add the repo unless the keys are added successfully.

Feel free to reopen if the problem still exists.
